src/rt-app_utils.c

```diff
@@ -321,3 +332,15 @@ void ftrace_write(int mark_fd, const char *fmt, ...)
 	}
 
 }
+
+int
+str2loglevel(char *s)
+{
+	for (int n=0; lls[n].val != LOG_LEVEL_END; n++) {
+		if (!strcmp(lls[n].s, s)) {
```

Comment:
always prefer strncmp to strcmp. strcmp relies on null terminating string which is often seen as a weakness
strncmp(lls[n].s, s, strlenght(lls[n].s))

Reply:
yes, for comparisons with user input, but here it seems a bit pointless - how would you end up with a string literal that isn't null-terminated? am i missing something?

Reply:
fair enough